unimus-backup-exporter.sh

```diff
@@ -0,0 +1,212 @@
+	fi
+}
+
+function get_allDevices (){
+	echoGreen "Getting Device Information" >> $log
+	for ((i=0; ; i+=1)); do
+
+		local contents=$(unimus_get "devices?page=$i")
+		
+		for((j=0; ; j+=1)); do
+			if ( jq -e '.data['$j'] | length == 0' <<< $contents) >/dev/null; then
+				break
+			fi
+			if $(echo "$contents" | ${devices[(jq -e ' .data['$j']')]}) >/dev/null; then
+				read -a value < <(echo $(jq -e '.data['$j'] | .id, .address' <<< $contents))
+				devices[${value[0]}]=$(echo ${value[1]}  | tr -d '"')
+			fi
+		done
+
+		if ( jq -e '.data | length == 0' <<< $contents ) >/dev/null; then
+			break
+		fi 
+	done
+}
+
+function get_all_backups (){
+	for key in "${!devices[@]}"; do
+		for ((i=0; ; i+=1));do
+			local contents=$(unimus_get "devices/$key/backups?page=$i")
+			for ((j=0; ; j+=1)); do
+				if  ( jq -e '.data['$j'] | length == 0' <<<  $contents) >/dev/null; then
+					break
+				fi
+
+				local deviceId=$key
+				local date="$(jq -e -r '.data['$j'].validSince' <<< $contents | { read tme ; date "+%F-%T-%Z" -d "@$tme" ; })"
+				local backup=$(jq -e -r '.data['$j'].bytes' <<< $contents)
+				local type=$(jq -e -r '.data['$j'].type' <<< $contents)
+				save_file "$deviceId" "$date" "$backup" "$type"
+			done
+		if [ $(jq -e '.data | length == 0' <<< $contents) ] >/dev/null; then
+				break
+		fi 
+		done
+	done
+}
+
+#Will Pull down backups and save to Disk
+function get_latest_backups (){
+
+	#Query for latest backups. This will loop through getting every page
+
+	for ((i=0; ; i+=1)); do
+		local contents=$(unimus_get "devices/backups/latest?page=$i")
+
+		for ((j=0; ; j+=1)); do
+
+			#Breaks if looped through all devices
+			if  ( jq -e '.data['$j'] | length == 0' <<<  $contents) >/dev/null; then
+				break
+			fi
+
+			local deviceId=$(jq -e -r '.data['$j'].deviceId' <<< $contents)
+			local date="$(jq -e -r '.data['$j'].backup.validSince' <<< $contents | { read tme ; date "+%F-%T-%Z" -d "@$tme" ; })"
+			local backup=$(jq -e -r '.data['$j'].backup.bytes' <<< $contents)
+			local type=$(jq -e -r '.data['$j'].backup.type' <<< $contents)
+
+			save_file "$deviceId" "$date" "$backup" "$type"
+
+		done
+
+		#breaks if empty page.
+		if [ $(jq -e '.data | length == 0' <<< $contents) ] >/dev/null; then
+			break
+		fi 
+	done
+}
+
+function pushToGit(){
+	cd $backup_dir
+	if ! [ "$(git rev-parse --is-inside-work-tree 2>/dev/null)" ]; then
+		git init 
+		git add . 
+		git commit -m "Initial Commit"
+		case $git_server_protocal in 
+			ssh)
+			git remote add orgin ssh://git@$git_server_address/$git_repo_name 
+			;;
+			http)
+			git remote add orgin http://$git_username:$git_password@$git_server_address:$git_port/$git_repo_name 
+			;;
+			https)
+			git remote add orgin https://$git_username:$git_password@$git_server_address:$git_port/$git_repo_name 
+			;;
+			*)
+			echoGreen "Invalid setting for git_server_protocal" 
+			;;
+		esac
+
+		git push -u orgin master
```

Comment:
use branch from config file instead of `master`

Reply:
Script now uses config file's branch variable instead of hardcoded master.